app/assets/javascripts/discourse/components/d-button.js.es6

```diff
@@ -23,6 +23,11 @@ export default Component.extend({
 
   btnIcon: Ember.computed.notEmpty("icon"),
 
+  @computed("class")
```

Comment:
This is a bit odd to me - is the idea where that you don't want `btn` if `btn-link` is present?

I think checking `class` is awkward for a couple reasons. It is a reserved word and it is confusing to pair it with classNameBinding because you get somewhat of a circular relationship.

Instead it would be better I think to pass type="link" and use that property to remove `btn` and andd `btn-link`

Reply:
Yep, that's the idea: using `<a>` tags as buttons is bad for accessibility (you have to explicitly give it a role of button, and even then some browsers won't allow keyboard interaction with it), so it's far better to use an actual `<button>` but then style it as a link. So `btn-link` does that styling and I don't want the `btn` class intefering.

Yeah, I hadn't caught that we have a circular dependency here - I'm amazed ember didn't blow up...

Reply:
I decided to use a new `display` attribute rather than using `type` and then having to deal with setting that to `type="button"` in the actual rendered `<button>`